patches/minecraft/net/minecraft/data/tags/TagsProvider.java.patch

```diff
@@ -9,21 +9,25 @@
 +   protected final net.minecraftforge.common.data.ExistingFileHelper existingFileHelper;
 +   private final net.minecraftforge.common.data.ExistingFileHelper.IResourceType resourceType;
  
-+   /**@deprecated Forge: Use the ModID version**/ @Deprecated
++   /**
++    * @see #TagsProvider(DataGenerator, Registry, String, net.minecraftforge.common.data.ExistingFileHelper)
++    * @see #TagsProvider(DataGenerator, Registry, String, net.minecraftforge.common.data.ExistingFileHelper, String)
++    * @deprecated Forge: Use one of the mod id variants
++    */
++   @Deprecated
     protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_) {
 +      this(p_126546_, p_126547_, "vanilla", null);
 +   }
-+   protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_, String modId, @javax.annotation.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper) {
++   protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_, String modId, @org.jetbrains.annotations.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper) {
 +      this(p_126546_, p_126547_, modId, existingFileHelper, null);
 +   }
-+   protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_, String modId, @javax.annotation.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper, @javax.annotation.Nullable String folder) {
++   protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_, String modId, @org.jetbrains.annotations.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper, @org.jetbrains.annotations.Nullable String folder) {
```

Comment:
Is this constructor with the folder being passed in just being kept for backward compat? As it seems to be unneeded now that `TagManager.getTagDir` exists and supports custom registries properly as of this PR? If so it should either be removed (as tags got broken so much and it isn't worth maintaining until 1.19 as people will have to adjust anyway) or it should be deprecated and the one without the folder passed in be preferred.

In fact I think what probably should happen is:
- remove this constructor (just have the one without the folder passed in)
- remove the class level `folder` variable as it isn't needed anymore
- Don't substring the `tags/` out of the returned folder from `getTagDir` and just pass it directly to the `ResourceType` constructor without prefixing it with `tags/` there in case any tag directories ever end up in a different subfolder